roles/vivumlab_wireguard/tasks/main.yml

```
@@ -1,19 +1,37 @@
 ---
-- name: Add the WireGuard PPA repository
+- name: Add the WireGuard PPA repository (Debian)
   apt_repository:
-    repo: ppa:wireguard/wireguard
+    repo: deb http://deb.debian.org/debian buster-backports main
     state: present
-  register: result
-  until: result is succeeded
-  retries: 10
-  delay: 3
+    update_cache: yes
+  when: (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] == "10")
```

```suggestion
  when: (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] == "10") or (ansible_facts['distribution'] == "Debian" and ansible_facts[distribution_major_version] == "11")
```
You excluded Version 11 in the other one, shouldn't this include 11 then? Just a guess, I'm usually using Ubuntu.

---

For buster backports? No
Buster-backports are programs/files whatever the exist in the new er version, but might be needed in a previous version.
So they 'backport' to 'buster'.

These files exist in bullseye (Version 11 of debian) already.